sources/packages/web/src/services/DesignationAgreementService.ts

```diff
@@ -32,9 +35,35 @@ export class DesignationAgreementService {
     );
   }
 
-  public async getDesignationsAgreements(): Promise<
-    GetDesignationAgreementsDto[]
-  > {
-    return ApiClient.DesignationAgreement.getDesignationsAgreements();
+  public async getDesignationsAgreements(
+    institutionId?: number,
+  ): Promise<GetDesignationAgreementsDto[]> {
+    return ApiClient.DesignationAgreement.getDesignationsAgreements(
+      institutionId,
+    );
+  }
+
+  public async getDesignationByStatus(
+    designationStatus: DesignationAgreementStatus,
+    searchCriteria?: string,
+  ): Promise<PendingDesignationDto[]> {
+    return ApiClient.DesignationAgreement.getDesignationByStatus(
+      designationStatus,
+      searchCriteria,
+    );
+  }
+
+  public async updateDesignationAgreement(
+    designationId: number,
+    designation: UpdateDesignationDto,
+  ): Promise<void> {
+    designation.locationsDesignations = designation.locationsDesignations?.filter(
```

Comment:
What would be the scenario where designation.locationsDesignations will be null?

Reply:
Comments updated for null check.